src/dns_internal.cc

```diff
@@ -1783,6 +1790,13 @@ idnsALookup(const char *name, IDNSCB * callback, void *data)
 
     if (q->do_searchpath && nd < ndots) {
         q->domain = 0;
+        const auto need = strlen(q->name) + 1 + strlen(searchpath[q->domain].domain);
+        if (need > NS_MAXDNAME) {
+            debugs(23, DBG_IMPORTANT, "SECURITY ALERT: searchpath name for '" << q->name << "' too long.");
+            idnsCallbackOnEarlyError(callback, data, "searchpath name too long");
```

Comment:
Please do not blame the searchpath entry for the problem. It is the _combination_ that is problematic. We do not know whether the searchpath entry or the requested domain are problematic in isolation. This change request applies to PR title and messages in code.

Reply:
Thank you! Just FYI: any reviews that I have not dealt with yet, I will get to; I will not ignore them, it's just I am waiting for other PRs to merge before working on these ones.

Reply:
apparently I have lied and I really don't have the time to go through all of the PRs anymore :(
feel free to close/convert-to-bugs any of them

Reply:
> That is unfortunate, but thank you for letting us know!
> 
> 
> 
> Instead of closing these unfinished PRs and creating bug reports (creating even more work for us and making important past context/comments more difficult to use), I propose to keep them open and nibble at them when others find the time to do so. I will help with a few. We can even use an existing `S-waiting-for-more-authors` label to mark them specially if needed. @kinkie, @yadij, any better ideas?

No. Thanks for bringing them so far Joshua, we'll take it from here

Reply:
Just to avoid a misunderstanding, I am working on some of the existing outstanding Joshua's PRs, and I believe we can process them all in fairly reasonable time, but I cannot volunteer to audit the list of URLs in https://github.com/squid-cache/squid/pull/2177#discussion_r2483151599 and create new PRs or bug reports for those changes that have not been posted as PRs yet.